compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InterfaceDelegationLowering.kt

```diff
@@ -201,3 +201,25 @@ private fun IrSimpleFunction.isDefinitelyNotDefaultImplsMethod() =
             (name.asString() == "clone" &&
                     parent.safeAs<IrClass>()?.fqNameWhenAvailable?.asString() == "kotlin.Cloneable" &&
                     valueParameters.isEmpty())
+
+internal val interfaceObjectCallsPhase = makeIrFilePhase(
+    lowering = ::InterfaceObjectCallsLowering,
+    name = "InterfaceObjectCalls",
+    description = "Resolve calls to Object methods on interface types to virtual methods"
+)
+
+private class InterfaceObjectCallsLowering(val context: JvmBackendContext) : IrElementTransformerVoid(), FileLoweringPass {
+    override fun lower(irFile: IrFile) = irFile.transformChildrenVoid(this)
+
+    override fun visitCall(expression: IrCall): IrExpression {
+        if (expression.superQualifierSymbol != null)
```

Comment:
Seems this cases also should be supported:
```
interface Base {

}

interface Derived: Base {
    fun foo() {
        super<Base>.hashCode()
    }
}
```
Actually compilation is failed in `InterfaceSuperCallsLowering`, but it could be simply fixed by smth like:
```
override fun visitCall(expression: IrCall): IrExpression {
        if (expression.superQualifierSymbol?.owner?.isInterface != true || expression.isSuperToAny()) {
            return super.visitCall(expression)
        }
```

Reply:
I've added code to support the case where `Derived` is a class, but I'm pretty sure it's not possible to make your example work. The problem is that to call `Object.hashCode` precisely `foo` needs an INVOKESPECIAL, but `foo` is moved to `Derived$DefaultImpls` which cannot do INVOKESPECIAL on instances of `Derived`, and it's not possible to put an accessor into `Derived` because it's an interface. If you actually try to load `Derived` from your example after compiling it with the non-IR backend, you'll get a validation error.

Reply:
Yes, really such cases seems should be prohibited on both targets (except calls within @JvmDefault's methods)

Reply:
https://youtrack.jetbrains.com/issue/KT-35417